sycl/doc/extensions/proposed/sycl_ext_oneapi_group_load_store_extended.asciidoc

```diff
@@ -0,0 +1,299 @@
+----
+namespace sycl::ext::oneapi::experimental {
+
+// Load API sycl::vec overload
+template<typename Group, typename InputIteratorT,
+         typename OutputT, int N, typename Properties = properties<>>
+void group_load(Group g, InputIteratorT in_ptr,
+                      sycl::vec<OutputT, N>& out, Properties = {}); // (3)
+
+// Store API sycl::vec overload
+template<typename Group, typename InputT, int N,
+         typename OutputIteratorT, typename Properties = properties<>>
+void group_store(Group g, const sycl::vec<InputT, N>& in,
+                      OutputIteratorT out_ptr, Properties = {}); // (4)
+
+}
+----
+
+(3) _Preconditions_: `in_ptr` must be the same for all work-items
+in the group.
+
+_Constraints_: Only available if `Group` is a work-group or sub-group.
+
+_Effects_: Loads `N` elements from `in_ptr` to `out`
+using the `g` group object. Properties can specify data placement.
+
+(4) _Preconditions_: `out_ptr` must be the same for all work-items
+in the group.
+
+_Constraints_: Only available if `Group` is a work-group or sub-group.
+
+_Effects_: Stores `N` elements from `in` vec to `out_ptr`
+using the `g` group object. Properties can specify data placement.
+
+== Functions with fixed-size arrays
+
+This API provides the following features:
+
+* load/store with temporary memory buffer, which can be passed via GroupHelper
+such as `group_with_scratchpad` link:https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/proposed/sycl_ext_oneapi_group_sort.asciidoc#group-helper[group_with_scratchpad description from sort over group proposal]
+
+* load/store flexible amount of elements per work item as sycl::span
+
+* specify data placement type via properties (6) or extra options such as
+setting boundary values or limiting numbers of work items (TODO in design
+considerations)
+
+[source,c++]
+----
+namespace sycl::ext::oneapi::experimental {
+
+// Load API
+template<typename GroupHelper, typename InputIteratorT, typename OutputT,
+        std::size_t ElementsPerWorkItem, typename Properties = properties<>>
+void group_load(GroupHelper gh, InputIteratorT in_ptr,
+                sycl::span<OutputT, ElementsPerWorkItem> out, Properties = {}); // (6)
+
+
+// Store API
+template<typename GroupHelper, typename OutputIteratorT, typename InputT,
+        std::size_t ElementsPerWorkItem, typename Properties = properties<>>
+void group_store(GroupHelper gh, sycl::span<InputT, ItemsPerWorkItem> in,
+                 OutputIteratorT out_ptr,  Properties = {}); // (7)
+
+
+// Properties:
+enum class group_algorithm_data_placement {
+  blocked,
+  striped
+};
+
+// Support memory function to define the needed amount of temporary memory
+// needed (name TBD)
+
+template<typename T, std::size_t ElementsPerWorkItem>
+constexpr std::size_t memory_required(sycl::memory_scope scope,
+                                      std::size_t block_size); // (8)
+
+}
+----
+
+(6) _Preconditions_: `in_ptr` must be the same for all work-items
```

Comment:
It seems like for this overload we actually do have properties we want to support: do we want to restrict the API to them? Like specifying which exact properties are supported by certain overloads? And what happens if wrong properties are passed?